FreeRTOS/Source/tasks.c

```diff
@@ -4676,13 +4730,21 @@ TickType_t uxReturn;
 
 #if ( configUSE_MUTEXES == 1 )
 
+#if ( configMUTEX_MULTI_STEP_PRIORITY_DISINHERITANCE == 0 )
 	TaskHandle_t pvTaskIncrementMutexHeldCount( void )
+#else
+	TaskHandle_t pvTaskIncrementMutexHeldCount( ListItem_t * xListItem )
+#endif
 	{
 		/* If xSemaphoreCreateMutex() is called before any tasks have been created
 		then pxCurrentTCB will be NULL. */
 		if( pxCurrentTCB != NULL )
 		{
 			( pxCurrentTCB->uxMutexesHeld )++;
+#if ( configMUTEX_MULTI_STEP_PRIORITY_DISINHERITANCE == 1 )
+			listSET_LIST_ITEM_VALUE(xListItem, ( ( TickType_t ) configMAX_PRIORITIES - ( TickType_t ) pxCurrentTCB->uxPriority ));

```

Comment:
When a task acquires a mutex, it stores the value of `( TickType_t ) configMAX_PRIORITIES - ( TickType_t ) pxCurrentTCB->uxPriority` into the `xMutexesHeld` list. 
Why not just `pxCurrentTCB->uxPriority`? 

In effect, the list stored by the task isnt the mutex objects themselves, but rather the (potentially inherited) priority of the task when it acquired the mutex?

Reply:
Because we want the highest priority entry to be on top (first) as opposed to last.

> In effect, the list stored by the task isnt the mutex objects themselves, but rather the (potentially inherited) priority of the task when it acquired the mutex?

It contains the reference to the mutex structure through the `owner` field, but we are not directly accessing it. But yes, we are mainly using the priorities, which are adjusted whenever these mutexes are attempted to be acquired, released or there is a timeout waiting on it.

Reply:
Ok, that makes sense, I can see it makes it easy to call this when disinheriting priorities and finding the next priority to assume
```
newPriority = ( UBaseType_t ) configMAX_PRIORITIES - ( UBaseType_t )listGET_ITEM_VALUE_OF_HEAD_ENTRY( &pxTCB->xMutexesHeld );
```